src/Core/DevOps/Environment/EnvironmentHelper.php

```diff
@@ -14,11 +19,57 @@ class EnvironmentHelper
      */
     public static function getVariable(string $key, $default = null)
     {
-        return $_SERVER[$key] ?? $_ENV[$key] ?? $default;
+        $value = $_SERVER[$key] ?? $_ENV[$key] ?? null;
+        $transformerData = new EnvironmentHelperTransformerData($key, $value, $default);
+
+        foreach (self::$transformers as $transformers) {
+            foreach ($transformers as $transformer) {
+                if (!is_subclass_of($transformer, EnvironmentHelperTransformerInterface::class, true)) {
+                    continue;
+                }
+
+                $transformer::transform($transformerData);
+            }
+        }
+
+        return $transformerData->getValue() ?? $transformerData->getDefault();
     }
 
     public static function hasVariable(string $key): bool
     {
-        return \array_key_exists($key, $_SERVER) || \array_key_exists($key, $_ENV);
+        return array_key_exists($key, $_SERVER) || array_key_exists($key, $_ENV);
+    }
+
+    public static function addTransformer(string $transformerClass, int $priority = 0): void
+    {
+        if (!is_subclass_of($transformerClass, EnvironmentHelperTransformerInterface::class, true)) {
```

Comment:
why not type hinting the parameter?

Reply:
@mitelg Typehinting would require the class to be instantiated but currently I am just using a static method only, so i cannot really typehint.

But I could change it to use a class instance instead.

Reply:
Let me know, if I should change it accordingly to use an instance.

Reply:
never mind, didn't see that you get the class string and the interface has a static method 😁🙈

all fine 👌